tests/IntegrationTests/ItemPropertiesSetupTest.php

```diff
@@ -0,0 +1,122 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\Matej\IntegrationTests;
+
+use Lmc\Matej\Model\Command;
+
+/**
+ * @covers ItemPropertiesSetupRequestBuilder
+ * @covers EventsRequestBuilder
+ */
+class ItemPropertiesSetupTest extends IntegrationTestCase
+{
+    /**
+     * In Matej's component MGW, this will throw an error, because the property isn't defined.
+     * The API will however return OK. This is desired behaviour.
+     *
+     * @test
+     */
+    public function shouldIssueEventWithUnknownPropertyButPassOk(): void
+    {
+        $response = $this->createMatejInstance()
+            ->request()
+            ->events()
+            ->addItemProperty(
+                Command\ItemProperty::create(
+                    'integration-test-php-clientitem-id',
+                    ['integration_test_php_client_property_1' => true, 'integration_test_php_client_property_2' => false]
+                )
+            )
+            ->send();
+
+        $this->assertSame(1, $response->getNumberOfCommands());
+        $this->assertSame(1, $response->getNumberOfSuccessfulCommands());
+        $this->assertSame(0, $response->getNumberOfFailedCommands());
+        $this->assertSame(0, $response->getNumberOfSkippedCommands());
+
+        $commandResponse = $response->getCommandResponses()[0];
+        $this->assertSame('OK', $commandResponse->getStatus());
+        $this->assertSame([], $commandResponse->getData());
+        $this->assertSame('', $commandResponse->getMessage());
+    }
+
+    /**
+     * @test
+     * @depends shouldIssueEventWithUnknownPropertyButPassOk
+     */
+    public function shouldCreateNewPropertiesInMatej(): void
+    {
+        $response = $this->createMatejInstance()
+            ->request()
+            ->setupItemProperties()
+            ->addProperty(Command\ItemPropertySetup::boolean('integration_test_php_client_bool'))
+            ->addProperty(Command\ItemPropertySetup::double('integration_test_php_client_double'))
+            ->addProperty(Command\ItemPropertySetup::int('integration_test_php_client_int'))
+            ->addProperty(Command\ItemPropertySetup::string('integration_test_php_client_string'))
+            ->addProperties([
+                Command\ItemPropertySetup::timestamp('integration_test_php_client_timestamp'),
+                Command\ItemPropertySetup::set('integration_test_php_client_set'),
+            ])
+            ->send();
+
+        $this->assertSame(6, $response->getNumberOfCommands());
+        $this->assertSame(6, $response->getNumberOfSuccessfulCommands());
+        $this->assertSame(0, $response->getNumberOfFailedCommands());
+        $this->assertSame(0, $response->getNumberOfSkippedCommands());
```

Comment:
Too bad there is no "get item properties" endpoint in Matej - it would allow to write end-to-end integration tests, where you check the actual result in the database 🤷‍ .

Reply:
RAD-720 created :)